File: 287-FindTheDuplicateNumber/287-FindTheDuplicateNumber.js
// Last updated: 4/6/2026, 5:56:42 PM
var findDuplicate = function(nums) {
    let slow = nums[0];
    let fast = nums[0]; 
    
    do {
        slow = nums[slow];
        fast = nums[nums[fast]];
    } while (slow !== fast);
 
    slow = nums[0];

    while (slow !== fast) {
        slow = nums[slow];
        fast = nums[fast];
    }

    return slow;
};

File: 287-FindTheDuplicateNumber/287-FindTheDuplicateNumber.cpp
// Last updated: 4/6/2026, 5:56:39 PM
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        unordered_set<int> s;

        for (int num : nums) {
            if (s.count(num)) return num;
            s.insert(num);
        }
        return -1;
    }
};